### PR TITLE
jq 1.4 JSON Processor

### PR DESCRIPTION
**Pull request by jamesmacwhite**
A useful tool for cutting and slicing JSON data directly within shell.

I found it useful for my needs, as it saves having to use crazy `sed` commands to filter output

### CODE REVIEW

File: addons/tools/jq/package.mk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#  along with OpenELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="jq"
+PKG_VERSION="1.4"
+PKG_REV="1"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE="http://stedolan.github.io/jq/"
+PKG_URL="https://github.com/stedolan/$PKG_NAME/archive/$PKG_NAME-$PKG_VERSION.zip"
```
Comment (stefansaraev) [inline addons/tools/jq/package.mk]: PKG_URL="http://stedolan.github.io/jq/download/source/$PKG_NAME-$PKG_VERSION.tar.gz" and delete unpack() below.

File: addons/tools/jq/package.mk [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+PKG_DEPENDS_TARGET="flex:host bison:host"
```
Comment (stefansaraev) [inline addons/tools/jq/package.mk]: PKG_DEPENDS_TARGET="toolchain"

**Comment by stefansaraev** [comment]
this seems to be a "partial" work. you must add:

1. addon() { ... } to package.mk

2.
```
PKG_IS_ADDON="yes"
PKG_ADDON_TYPE="xbmc.python.script"
PKG_ADDON_PROVIDES=""
PKG_ADDON_REPOVERSION="6.0"
```
to package.mk

3. changelog.txt. icon/icon.png, a dummy source/default.py. see others in addons/tools/

File: addons/tools/jq/patches/jq-1.4_Makefile.patch [hunk 3]
```diff
@@ -0,0 +1,20 @@
+--- jq-1.4/Makefile.am	2014-08-17 18:35:48.139160558 -0400
++++ jq-1.4.patch/Makefile.am	2014-08-17 18:36:23.703159881 -0400
+@@ -34,7 +34,7 @@
+ 
+ lib_LTLIBRARIES = libjq.la
+ libjq_la_SOURCES = ${LIBJQ_SRC}
+-libjq_la_LIBADD = -lm
++libjq_la_LIBADD = -lm -lcrypto
```
Comment (stefansaraev) [inline addons/tools/jq/patches/jq-1.4_Makefile.patch]: if -lcrypto is hard dependency, libressl should be added to PKG_DEPENDS_TARGET

**Comment by jamesmacwhite** [comment]
Hi,

Thanks for the feedback. I'm rather new the OpenELEC build system, so apologies for the mistakes.

I will update the PR with the required amends you have highlighted.

**Comment by stefansaraev** [comment]
> Thanks for the feedback. I'm rather new the OpenELEC build system, so apologies for the mistakes

np :)

hint: use scripts/create_addon to make an "installable" zip

File: addons/tools/jq/package.mk (hunk 1 above)
Comment (jamesmacwhite) [inline addons/tools/jq/package.mk]: Without the unpack function the path becomes:

`$BUILD/jq/jq-jq-1.4`

Is there a better way of handling it?

Comment (stefansaraev) [inline addons/tools/jq/package.mk]: not if you use the above mentioned PKG_URL

Comment (jamesmacwhite) [inline addons/tools/jq/package.mk]: Ah! Sorry I misinterpreted your response as two separate issues, but see why they are related now.

**Comment by jamesmacwhite** [comment]
Updated the package.mk to the requirements requested. Marked as Rev 2. The license was incorrectly labelled which has also been updated.

Based on further discussion on IRC, the SSL dependency is questionable, so the patches have been removed.

I hope the update PR, is to the requirements of the OpenELEC team.

**Comment by stefansaraev** [comment]
thanks much. I'll do build test for all platforms (x86_64 is fine so far) and merge it later.

**Comment by jamesmacwhite** [comment]
Great, thanks very much for assisting with the PR!

**Pull request merged**